### PR TITLE
Add dependencies to all parts of the property ref path in path expand

### PR DESCRIPTION
**Pull request by gordon-klotho**
Since `lambda#Subnets` had 1 value from another path expansion, the other expansion wasn't properly depending on the final resolution of the subnets, so when `Expand(aws:lambda_function:lambda_test_app -> aws:efs_mount_target:test-efs-fs:lambda_test_app-test-efs-fs, <network>, source#Subnet#AvailabilityZone)` ran before, the lambda only had 1 subnet and wasn't properly setting up the path to mount point for the other subnet.

Pending e2e test to confirm, but resources.yaml now has 2 mount_points which it did not previously.

### CODE REVIEW

File: pkg/engine2/operational_eval/vertex_path_expand.go [hunk 1]
```diff
@@ -471,15 +472,8 @@ func (v *pathExpandVertex) Dependencies(eval *Evaluator) (graphChanges, error) {
...
-	var errs error
-	if propertyReferenceInfluencesEdge(v.Satisfication.Source) {
-		keys := getDepsForPropertyRef(eval.Solution, v.Edge.Source, v.Satisfication.Source.PropertyReference)
-		changes.addEdges(srcKey, keys)
-	}
-	if propertyReferenceInfluencesEdge(v.Satisfication.Target) {
-		keys := getDepsForPropertyRef(eval.Solution, v.Edge.Target, v.Satisfication.Target.PropertyReference)
-		changes.addEdges(srcKey, keys)
-	}
+	changes.addEdges(srcKey, getDepsForPropertyRef(eval.Solution, v.Edge.Source, v.Satisfication.Source.PropertyReference))
```
Comment (gordon-klotho) [inline pkg/engine2/operational_eval/vertex_path_expand.go:475]: This was 1 part of the fix - the dependencies were getting skipped due to it having Validity.

File: pkg/engine2/operational_eval/vertex_path_expand.go [hunk 2]
```diff
@@ -584,16 +579,20 @@ func getDepsForPropertyRef(
...
+		return nil
+	}
 	keys := make(set.Set[Key])
 	cfgCtx := solution_context.DynamicCtx(sol)
 	currResources := []construct.ResourceId{res}
 	parts := strings.Split(propertyRef, "#")
 	for _, part := range parts {
 		var nextResources []construct.ResourceId
 		for _, currResource := range currResources {
+			keys.Add(Key{Ref: construct.PropertyRef{Resource: currResource, Property: part}})
```
Comment (gordon-klotho) [inline pkg/engine2/operational_eval/vertex_path_expand.go:592]: This is the other part of the fix, all elements of the property path need to be depended on.

File: pkg/engine2/path_selection/path_expansion.go [hunk 3]
```diff
@@ -128,23 +135,45 @@ func renameAndReplaceInTempGraph(
...
+				continue
+			}
+
+		case err != nil:
+			errs = errors.Join(errs, err)
+			continue
+		}
+		result[i] = r
+		if i > 0 {
+			err = g.AddEdge(result[i-1].ID, r.ID)
```
Comment (gordon-klotho) [inline pkg/engine2/path_selection/path_expansion.go:165]: Not sure why my change broke the k8s functional test, but this change (`id` -> `r.ID`) fixed it.

**Comment by gordon-klotho** [comment]
e2e test passed
```
INFO:SuiteRunner:Test result for aws: name: aws
        result: Result.SUCCESS
```

**Review by jhsinger-klotho — APPROVED** [review]

**Pull request merged**